docs/gallery/advanced_io/streaming.py

```diff
@@ -23,113 +23,111 @@
+# creates a virtual filesystem for remote stores. With this approach, a virtual file is created for the file and
+# the virtual filesystem layer takes care of requesting data from the S3 bucket whenever data is
+# read from the virtual file.  Note that this implementation is completely unaware of internals of the HDF5 format
+# and thus can work for **any** file, not only for the purpose of use with H5PY and PyNWB.
+#
+# First install ``fsspec`` and the dependencies of the :py:class:`~fsspec.implementations.http.HTTPFileSystem`:
+#
+# .. code-block:: bash
+#
+#    pip install fsspec requests aiohttp
+#
+# Then in Python:
+
+import fsspec
+import pynwb
+import h5py
+from fsspec.implementations.cached import CachingFileSystem
+
+# first, create a virtual filesystem based on the http protocol
+fs = fsspec.filesystem("http")
+
+# create a cache to save downloaded data to disk (optional)
+fs = CachingFileSystem(
+    fs=fs,
+    cache_storage="nwb-cache",  # Local folder for the cache
+)
+
+# next, open the file
+with fs.open(s3_url, "rb") as f:
+    with h5py.File(f) as file:
+        with pynwb.NWBHDF5IO(file=file, load_namespaces=True) as io:
+            nwbfile = io.read()
+            print(nwbfile.acquisition['lick_times'].time_series['lick_left_times'].data[:])
+
+##################################
+# fsspec is a library that can be used to access a variety of different store formats, including (at the time of
+# writing):
+#
+# .. code-block:: python
+#
+#     from fsspec.registry import known_implementations
+#     known_implementations.keys()
+#
+# file, memory, dropbox, http, https, zip, tar, gcs, gs, gdrive, sftp, ssh, ftp, hdfs, arrow_hdfs, webhdfs, s3, s3a,
+# wandb, oci, adl, abfs, az, cached, blockcache, filecache, simplecache, dask, dbfs, github, git, smb, jupyter, jlab,
+# libarchive, reference
+#
+# The S3 backend, in particular, may provide additional functionality for accessing data on DANDI. See the
+# `fsspec documentation on known implementations <https://filesystem-spec.readthedocs.io/en/latest/api.html?highlight=S3#other-known-implementations>`_
+# for a full updated list of supported store formats.
+#
+# Streaming Method 2: ROS3
+# ------------------------
+# ROS3 is one of the supported methods for reading data from a remote store. ROS3 stands for "read only S3" and is a
+# driver created by the HDF5 Group that allows HDF5 to read HDF5 files stored remotely in s3 buckets. Using this method
+# requires that your HDF5 library is installed with the ROS3 driver enabled. This is not the default configuration,
+# so you will need to make sure you install the right version of ``h5py`` that has this advanced configuration enabled.
+# You can install HDF5 with the ROS3 driver from `conda-forge <https://conda-forge.org/>`_ using ``conda``. You may
+# first need to uninstall a currently installed version of ``h5py``.
+#
+# .. code-block:: bash
+#
+#    pip uninstall h5py
+#    conda install -c conda-forge "h5py>=3.2"
```

Comment:
Since this has been in h5py now for 2 years, could this be shortened to just say something along the lines of "ROS3 requires a recent version of h5py >= 3.2"

Reply:
Or is installing h5py with ROS3 enabled still and issue?